engine/lib/luajit-ffi-gen/src/impl_item/generate.rs

```diff
@@ -267,9 +272,16 @@ fn gen_func_body(self_ident: &Ident, method: &MethodInfo) -> TokenStream {
         };
 
         let method_call = if ty.is_option {
-            quote! {
-                #method_call
-                let Some(__res__) = __res__ else { return std::ptr::null(); };
+            if ty.is_mutable {
```

Comment:
@HaronK looks like you fixed the same thing in https://github.com/Limit-Theory-Redux/ltheory/pull/201/files#diff-b07fa8fd8254a410c8c9899fff6a102c0643785dac5687f7606c4508646abffeL269